libr/core/canal.c

```diff
@@ -1154,26 +1154,28 @@ static int core_anal_graph_nodes(RCore *core, RAnalFunction *fcn, int opts) {
 	} else if (is_json) {
 		// TODO: show vars, refs and xrefs
 		char *fcn_name_escaped = r_str_escape_utf8_for_json (fcn->name, -1);
-		r_cons_printf ("{\"name\":\"%s\"", r_str_get (fcn_name_escaped));
+		pj_o (pj);
+		pj_ks (pj, "name", r_str_get (fcn_name_escaped));
 		free (fcn_name_escaped);
-		r_cons_printf (",\"offset\":%"PFMT64d, fcn->addr);
-		r_cons_printf (",\"ninstr\":%"PFMT64d, (ut64)fcn->ninstr);
-		r_cons_printf (",\"nargs\":%d",
+		pj_kn (pj, "offset", fcn->addr);
+		pj_ki (pj, "ninstr", fcn->ninstr);
+		pj_ki (pj, "nargs",
```

Comment:
i dont see why number of args can be negative :?

Reply:
From a logical perspective you are completely right. However, i was usually looking at the type of the variables that are added to the json output. For instance `r_anal_var_count()` returns int anyway, so from my perspective it was a better approach to simply use that type.